chime/view_functions.py

```diff
@@ -435,10 +439,23 @@ def decorated_function(*args, **kwargs):
             return redirect(redirect_url)
 
         auth_data_href = current_app.config['AUTH_DATA_HREF']
-        if not is_allowed_email(get_auth_data_file(auth_data_href), email):
+        last_check = session.get('auth_check', {}).get('last_check', 0.0)
+        last_result = session.get('auth_check', {}).get('last_result')
+
+        if last_result is True and (last_check + AUTH_CHECK_LIFESPAN) > time.time():
+            Logger.info("Email still allowed.")
+
+        elif not is_allowed_email(get_auth_data_file(auth_data_href), email):
+            Logger.debug('Remembering that email was not allowed')
+            session['auth_check'] = dict(last_check=time.time(), last_result=False)
```

Comment:
It’s not; we simply cache state and then only change behavior on one particular value. I do think it’s important to cache the state in all cases.